File: src/main/java/models/Activite.java
package models;

public class Activite {
    int id;
    String nom_activite,date,heure,statut;
    public Activite() {}
    public Activite(int id, String nom_activite, String date, String heure, String statut) {
        this.id = id;
        this.nom_activite = nom_activite;
        this.date = date;
        this.heure = heure;
        this.statut = statut;

    }
    public Activite(String nom_activite, String date, String heure, String statut) {

        this.nom_activite = nom_activite;
        this.date = date;
        this.heure = heure;
        this.statut = statut;

    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getStatut() {
        return statut;
    }

    public void setStatut(String statut) {
        this.statut = statut;
    }

    public String getHeure() {
        return heure;
    }

    public void setHeure(String heure) {
        this.heure = heure;
    }

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public String getNom_activite() {
        return nom_activite;
    }

    public void setNom_activite(String nom_activite) {
        this.nom_activite = nom_activite;
    }

    @Override
    public String toString() {
        return "Activite{" +
                "id=" + id +
                ", nom_activite='" + nom_activite + '\'' +
                ", date='" + date + '\'' +
                ", heure='" + heure + '\'' +
                ", statut='" + statut + '\'' +
                '}';
    }
}

File: ProjetWebTrekSwap/src/Entity/Activite.php
<?php

namespace App\Entity;

use App\Repository\ActiviteRepository;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

#[ORM\Entity(repositoryClass: ActiviteRepository::class)]
class Activite
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(nullable: true)]
    #[Assert\NotNull(message: "L'ID de la destination ne peut pas être nul.")]
    private ?int $id_destination = 0;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message: "Le nom de l'activité ne peut pas être vide.")]
    #[Assert\Length(max: 255, maxMessage: "Le nom de l'activité ne peut pas dépasser {{ limit }} caractères.")]
    private ?string $nom_activite = null;

    #[ORM\Column(type: Types::DATE_MUTABLE)]
    #[Assert\NotNull(message: "La date de l'activité ne peut pas être nulle.")]
    #[Assert\Range(
        min: "today",
        minMessage: "La date de l'activité doit être après aujourd'hui.",
        invalidMessage: "La date doit être au format valide."
    )]
    private ?\DateTimeInterface $date = null;
    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message: "L'heure ne peut pas être vide.")]
    #[Assert\Length(max: 255, maxMessage: "L'heure ne peut pas dépasser {{ limit }} caractères.")]
    private ?string $heure = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message: "Le statut de l'activité ne peut pas être vide.")]
    #[Assert\Choice(choices: ['active', 'inactive','completed'], message: "Le statut doit être 'active' ou 'inactive' ou 'Completed' .")]
    private ?string $statut = null;
    public function getId(): ?int
    {
        return $this->id;
    }

    public function getIdDestination(): ?int
    {
        return $this->id_destination;
    }

    public function setIdDestination(?int $id_destination): void
    {
        $this->id_destination = $id_destination;
    }

    public function getNomActivite(): ?string
    {
        return $this->nom_activite;
    }

    public function setNomActivite(string $nom_activite): static
    {
        $this->nom_activite = $nom_activite;

        return $this;
    }

    public function getDate(): ?\DateTimeInterface
    {
        return $this->date;
    }

    public function setDate(\DateTimeInterface $date): static
    {
        $this->date = $date;

        return $this;
    }

    public function getHeure(): ?string
    {
        return $this->heure;
    }

    public function setHeure(string $heure): static
    {
        $this->heure = $heure;

        return $this;
    }

    public function getStatut(): ?string
    {
        return $this->statut;
    }

    public function setStatut(string $statut): static
    {
        $this->statut = $statut;

        return $this;
    }
}
